lib/i18n/tasks/data/tree/siblings.rb

```diff
@@ -228,6 +228,14 @@ def add_ancestors_that_only_contain_nodes!(nodes)
       end
     end
 
+    def conditionally_warn_add_children_to_leaf(node, children)
+      return unless @warn_about_add_children_to_leaf
+      return if !children.empty? &&
+                (children.map(&:key) - I18n::Tasks::PluralKeys::CLDR_CATEGORY_KEYS).empty?
```

Comment:
Consider exposing and reusing the following method here:

https://github.com/glebm/i18n-tasks/blob/774c3d6259f199e3fd3ddf104beb101d8089a81e/lib/i18n/tasks/plural_keys.rb#L50

Reply:
Good suggestion! Refactored and added extra test to cover "internal node" situation